Sources/main.swift

```diff
@@ -0,0 +1,75 @@
+import HTTPServer
+import Router
+import JSON
+
+func getIdParameter(from request: Request) -> Int {
+    guard let idString = request.pathParameters["id"] else { return 0 }
+    return Int(idString) ?? 0
+}
+
+let router = Router { route in
+
+    var todos = [JSON?]()
+    let corsHeader = Headers(["access-control-allow-origin" : Header(stringLiteral: "*")])
+    
+    route.get("/") { _ in
+        var todooutput = ""
+        for (index, _todo) in todos.enumerated() {
+            if var todo = _todo where _todo != nil {
+                if index > 0 { todooutput += "," }
+                todooutput += JSONSerializer().serializeToString(todo)
+            }
+        }
+        return Response(body: Data(stringLiteral: "[\(todooutput)]"), headers: corsHeader)
+    }
+    
+    route.post("/") { request in
+        switch request.body {
+            case .buffer(let data):
+                var json = try JSONParser().parse(data)
+                json["url"] = JSON.from("http://localhost:34197/todo/\(todos.count)")
+                json["completed"] = JSON.from(false)
+                todos.append(json)
+                return Response(body:Data(JSONSerializer().serializeToString(json)), headers: corsHeader)
+            default: break
+        }
+        return Response(body:Data(), headers: corsHeader)
+    }
+    
+    route.delete("/") { request in
+        todos = [JSON?]()
+        return Response(headers: corsHeader)
+    }
+    
+    route.get("/todo/:id") { request in
```

Comment:
This is a great place to compose routers. Instead of repeating `route.method("/todo/:id")`, you can do
```
let todoRouter = Router(middleware: getIdParameter) { route in
    route.get("/") { ...
}
// inside other router
route.compose("/todo/:id", router: todoRouter)
```